operator/watchers/node_taint.go

```diff
@@ -457,6 +457,13 @@ func HandleNodeTolerationAndTaints(wg *sync.WaitGroup, clientset k8sClient.Clien
 	// that was initialized in nodesInit.
 	ciliumPodsWatcher(wg, clientset.Slim(), nodeQueue, stopCh, logger)
 
+	if option.Config.RemoveCiliumNodeTaints {
+		err := uninstallCiliumTaints(context.Background(), clientset, &nodeGetter{}, logger)
```

Comment:
As far as I understand, this code is in the setup path so it may be triggered even in regular usage, not just uninstall. I also see that the `checkTaintForNextNodeItem()` call below is already set up to check the `options.RemoveNodeTaint` option and react to it. I'm not sure what the solution should look like, but maybe could you speak a bit about how you validated that this fix works and that it doesn't create other unintended side effects under regular operation (ie not when uninstalling)?